backend/inspirehep/records/serializers/bibtex.py

```diff
@@ -16,6 +19,24 @@
 
 
 class BibtexWriter(Writer):
+    def _encode(self, text):
+        import latexcodec  # NOQA
+
+        if "$" not in text:
+            return codecs.encode(text, "ulatex+{}".format(self.encoding))
+        else:
+            found_math_expressions = [
+                f"{expression}" for expression in re.findall(r"\$.*\$", text)
+            ]
+            encoded_text = codecs.encode(text, "ulatex+{}".format(self.encoding))
+            for list_nb, _math_expression in enumerate(
```

Comment:
* Better to use [`zip`](https://docs.python.org/3.6/library/functions.html#zip) than fiddling with list indices for this kind of things. 
* Why the leading `_`?
* Nice trick, I had something much more complex in mind (namely splitting the string on math boundaries while keeping track of which mode every part is, and only encoding those that need encoding). Although it shouldn't happen, there's the risk that you get a different number of math segments in the non-encoded and encoded strings.
I would add an assertion like
```python
assert len(found_math_expressions) == len(encoded_math_expressions), f"number of math parts in LaTeX-encoded string is different for '{text}'."
```
before to make sure this doesn't happen.